### PR TITLE
chore: `/availability` - Remove Pages Router / Render App Router

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes CAL-4275
- Remove `/availability` routes from Pages Router and render them from App router.

### Tested

Screenshot 2024-12-12 at 10 33 51 PM

Screenshot 2024-12-12 at 10 33 38 PM

Screenshot 2024-12-12 at 10 33 44 PM

Screenshot 2024-12-12 at 10 32 47 PM

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Navigate to:

1. `/availability`
2.`/availability/troubleshoot`
3.`/availability/[schedule]`

### CODE REVIEW

**Review by Udit-takkar — DISMISSED** [review]

**Comment by keithwillcode** [comment]
@hbjORbj I see the note about "should not be merged until perf measures are done". Are you going to add that here?

**Comment by hbjORbj** [comment]
> @hbjORbj I see the note about "should not be merged until perf measures are done". Are you going to add that here?

@keithwillcode Hey Keith, I will actually add it in #16536 as that PR must bring quite perf improvements. I will turn this PR into a draft PR until then.

**Comment by hbjORbj** [comment]
DO NOT MERGE UNTIL EITHER (1) REVALIDATION LOGIC FOR SSR IS ADDED or (2) SSR IS REVERTED

**Comment by zomars** [comment]
Re-drafting due the [DO NOT MERGE YET] notice

File: apps/web/app/availability/page.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  try {
+    // const currentOrg = await OrganizationRepository.findCurrentOrg({
+    //   orgId,
+    //   userId,
+    // });
+    return (
+      <AvailabilityPage
+      //  currentOrg={currentOrg}
+      />
+    );
```
Comment (hbjORbj) [inline apps/web/app/availability/page.tsx:33]: commenting out sever-side fetching for now for defensive migration

File: apps/web/app/availability/[schedule]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  // } catch (e) {}
+
+  // try {
+  //   travelSchedules = await TravelScheduleRepository.findTravelSchedulesByUserId(userId);
+  // } catch (e) {}
+
+  return (
+    <AvailabilitySettingsWebWrapper
+
+    // scheduleFetched={schedule} travelSchedules={travelSchedules}
```
Comment (hbjORbj) [inline apps/web/app/availability/[schedule]/page.tsx:72]: commenting out sever-side fetching for now for defensive migration

**Review by emrysal — DISMISSED** [review]
💯

**Comment by hbjORbj** [comment]
@emrysal can you re-approve? I pushed a commit that adds availability routes to config.matcher in middleware

**Review by emrysal — APPROVED** [review]

**Pull request merged**